packages/engine-server/src/DendronEngineV3.ts

```diff
@@ -859,21 +855,169 @@ export class DendronEngineV3 extends EngineV3Base implements DEngine {
       });
     }
     this.logger.info({ ctx, msg: "exit" });
-    return {
-      data: notes,
-    };
+    return notes;
   }
 
-  async renderNote(): Promise<RenderNoteResp> {
-    throw Error("renderNote not implemented");
+  async renderNote({
+    id,
+    note,
+    flavor,
+    dest,
+  }: RenderNoteOpts): Promise<RenderNoteResp> {
+    const ctx = "DEngine:renderNote";
+
+    // If provided, we render the given note entirely. Otherwise find the note in workspace.
+    if (!note) {
+      note = (await this.getNote(id)).data;
+    } else {
+      // `procRehype` needs the note to be in the engine, so we have to add it in case it's a dummy note
```

Comment:
This actually shouldn't be necessary anymore since we stopped using the engine in `procRehype` - can we try removing this `this.writeNote` and see if it still works? 

If it's too much for now, maybe add a `TODO:` that captures this.